astropy/io/fits/column.py

```diff
@@ -1389,16 +1387,13 @@ def __getattr__(self, name):
 
         Implements for example self.units, self.formats, etc.
         """
-
         cname = name[:-1]
         if cname in KEYWORD_ATTRIBUTES and name[-1] == 's':
-            attr = []
-            for col in self:
-                val = getattr(col, cname)
-                if val is not None:
-                    attr.append(val)
-                else:
-                    attr.append('')
+            attr = [getattr(col, cname) for col in self.columns]
+            if None in attr:
```

Comment:
I'm not sure which is the optimal way, I personally wouldn't do a list comprehension followed by a contains iteration and then manually iterate over it again. It's also hard to time which approach is fastest but I'd rather remove that ``if None in attr`` and immediatly iterate over the ``enumerate(attr)`` or make it one-line with a generator-ternary-list-comprehension:
```
attr = ['' if colname is None else colname for colname in (getattr(col, cname) for col in self.columns)]
```
But it really depends what the most common case is which approach is fastest and I don't know if the last one is really all that PEP8 conform